src/pages/staticPages/tools/oncoprinter/OncoprinterStore.ts

```diff
@@ -0,0 +1,311 @@
+            const geneIds = this.hugoGeneSymbols.result!;
+            if (geneIds.length > 0) {
+                return _.keyBy(await client.fetchGenesUsingPOST({
+                    geneIdType: "HUGO_GENE_SYMBOL",
+                    geneIds
+                }), "hugoGeneSymbol");
+            } else {
+                return {};
+            }
+        }
+    });
+
+    readonly oncoKbAnnotatedGenes = remoteData({
+        invoke:()=>fetchOncoKbAnnotatedGenesSuppressErrors()
+    }, {});
+
+    readonly oncoKbData = remoteData<IOncoKbData|Error>({
+        await: () => [
+            this.nonAnnotatedGeneticData,
+            this.oncoKbAnnotatedGenes
+        ],
+        invoke: async() => {
+            if (AppConfig.serverConfig.show_oncokb) {
+                let result;
+                try {
+                    result = await fetchOncoKbDataForMutations(this.oncoKbAnnotatedGenes.result!, this.nonAnnotatedGeneticData.result!);
+                } catch(e) {
+                    result = new Error();
+                }
+                return result;
+            } else {
+                return ONCOKB_DEFAULT;
+            }
+        },
+        onError: (err: Error) => {
+            // fail silently, leave the error handling responsibility to the data consumer
+        }
+    }, ONCOKB_DEFAULT);
+
+    readonly oncoKbCnaData = remoteData<IOncoKbData|Error>({
+        await: () => [
+            this.nonAnnotatedGeneticData,
+            this.oncoKbAnnotatedGenes
+        ],
+        invoke: async() => {
+            if (AppConfig.serverConfig.show_oncokb) {
+                let result;
+                try {
+                    result = await fetchOncoKbDataForCna(this.oncoKbAnnotatedGenes.result!, this.nonAnnotatedGeneticData.result!);
+                } catch(e) {
+                    result = new Error();
+                }
+                return result;
+            } else {
+                return ONCOKB_DEFAULT;
+            }
+        },
+        onError: (err: Error) => {
+            // fail silently, leave the error handling responsibility to the data consumer
+        }
+    }, ONCOKB_DEFAULT);
+
+    readonly cbioportalCountData = remoteData<{[mutationPositionKey:string]:number}>({
+        await: ()=>[
+            this.nonAnnotatedGeneticData
+        ],
+        invoke: async()=>{
+
+            const mutations =
+                this.nonAnnotatedGeneticData.result!.filter(
+                    x=>(x.proteinPosStart !== undefined && x.proteinPosEnd !== undefined)
+                ) as Pick<Mutation, "entrezGeneId"|"proteinPosStart"|"proteinPosEnd">[];
+
+            const mutationPositionIdentifiers = _.values(countMutations(mutations));
+
+            const data = await client.fetchMutationCountsByPositionUsingPOST({
+                mutationPositionIdentifiers
+            });
+
+            return _.chain(data).groupBy(mutationCountByPositionKey).mapValues(counts=>_.sumBy(counts, "count")).value();
```

Comment:
again, i don't think TS checks validity of the key string.  unless i'm wrong, lets use (o)=>o.count).